### PR TITLE
[feature/OPT-1017] to dev

### PR DESCRIPTION
**Pull request by dantolin-iriusrisk**

### CODE REVIEW

**Comment by dantolin-iriusrisk** [comment]
@jmgarcia-iriusrisk @smaneroiriusrisk As this is a task not supervised by QA, we need both of you to accept the revision.

The SonarCloud analysis will fail because we are duplicating some code regarding slp_visio and we are not creating any tests for now. I will manually mark as accepted the results to be able to merge.

File: slp_drawio/resources/schemas/drawio_mapping_schema.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      "items": {
+        "type": "object",
+        "required": ["label"],
+        "properties": {
+          "label": {"$ref": "#/definitions/query"},
+          "type": {"$ref": "#/definitions/query"}
+        }
+      }
+    },
+    "dataflows": {
```
Comment (jmgarcia-iriusrisk) [inline slp_drawio/resources/schemas/drawio_mapping_schema.json:29]: In diagrams we won't map dataflow, IMHO this value should be removed and keep it the simpler.
It is already removed in DEV for slp_visio processor

File: slp_drawio/slp_drawio/load/drawio_mapping_file_loader.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class DrawioMapping:
+
+    def __init__(self, trustzones, components, dataflows):
+        self.mapping_trustzones = trustzones
+        self.mapping_components = components
+        self.mapping_dataflows = dataflows
```
Comment (jmgarcia-iriusrisk) [inline slp_drawio/slp_drawio/load/drawio_mapping_file_loader.py:15]: Remove this mapping_dataflow as it won't apply to this processor

File: slp_drawio/slp_drawio/load/drawio_mapping_file_loader.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            raise LoadingMappingFileError('Error loading the mapping file. The mapping files are not valid.',
+                                          e.__class__.__name__, str(e))
+
+    def __get_trustzones(self):
+        trustzone_mappings_list = jmespath.search("trustzones", self.merged_mappings)
+        return dict(zip([tz['label'] for tz in trustzone_mappings_list], trustzone_mappings_list))
+
+    def __get_components(self):
+        component_mappings_list = jmespath.search("components", self.merged_mappings)
+        return dict(zip([tz['label'] for tz in component_mappings_list], component_mappings_list))
```
Comment (jmgarcia-iriusrisk) [inline slp_drawio/slp_drawio/load/drawio_mapping_file_loader.py:43]: This logic (equals to Visio in MAIN branch) has been changed in DEV as it is not suitable for all the UC.
Review https://github.com/iriusrisk/startleft/blob/319ed1628d09975ac3d5d6ae535476f79f6d1e91/slp_visio/slp_visio/load/visio_mapping_loader.py#L41 to check if it is better for this case

File: slp_drawio/resources/schemas/drawio_mapping_schema.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        }
+      }
+    },
+    "components": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": ["label"],
+        "properties": {
+          "label": {"$ref": "#/definitions/query"},
```
Comment (jmgarcia-iriusrisk) [inline slp_drawio/resources/schemas/drawio_mapping_schema.json:24]: Should we be able to map components at least by String or List?

**Review by jmgarcia-iriusrisk — COMMENTED** [review]
Great work!!
I have written some comment that I'd like to discuss

**Review by jmgarcia-iriusrisk — APPROVED** [review]

**Review by smaneroiriusrisk — APPROVED** [review]

**Pull request merged**